### PR TITLE
Facility API Feature Update (POST add booking and GET booking)

### PR DESCRIPTION
**Pull request by ttheng3810**
Update the POST ```/facilities/bookings/``` (```add_bookings()```) such that 2 new JSON arguments are implemented, i.e. ```bookUntil``` and ```forceBooking``` (**updated**), as well as the GET ```bookings``` code segments.

*POST add booking:*
1. ```bookUntil``` (UNIX timestamp in integers): optional argument ~~which replaces ```repeat``` argument for the current version~~ (**Update**: which is an alternative to ```repeat``` [integer] argument); when specified, new bookings are placed recurrently (spaced a week apart) until any point of time within 1 week before the ```bookUntil``` timestamp.

1. ```forceBooking``` (boolean value): required argument; when set as a falsy value (e.g. ```0```), the entire booking request will be rejected if there is at least one conflicting existing booking placed; when set as a truthy value (e.g. ```1```), in case of recurrent bookings, new bookings will be booked if the booking lands on an unblocked slot.

*GET booking:*
For requests that return ```ccaID``` fields, ~~change to returning ```ccaName``` instead~~ (**Update**: add ```ccaName``` alongside).

### CODE REVIEW

**Review by angnobel — CHANGES_REQUESTED** [review]
Some comments

1. If bookUntil is used, return number of bookings made
2. If there is conflict bookings now, 500 error if 'forceBook' is not provided. If not provided should default to false
3. If forceBook is true, return number of successful bookings
4. dont need remove ccaID, just add ccaName

File: backend/FacilityBooking/FacilitiesAPI.py [hunk 1]
```diff
@@ -282,34 +298,77 @@ def add_booking():
...
+                            blocked_bookings[str(timestamp)].append(dict(conflict_bookings))
+
+                if len(insertData) != 0:
+                    db.Bookings.insert_many(insertData)
+                    response = {"message": "Unblocked slots booked", "blocked_bookings": blocked_bookings, "num_of_successful_bookings": str(len(insertData)), "status": "success"}
+                else:
+                    return make_response({"message": "All slots have been blocked.", "blocked_bookings": blocked_bookings, "status": "failed"}, 409)
+                
+
+
```
Comment (cjunxiang) [inline backend/FacilityBooking/FacilitiesAPI.py:365]: wah so many empty lines
@angnobel get them to lint leh hAHAHAHH

File: backend/FacilityBooking/FacilitiesAPI.py [hunk 2]
```diff
@@ -282,34 +298,77 @@ def add_booking():
...
+        elif (len(conflict) > 0):
+            if bool(formData["forceBook"]) == False:
+                return make_response({"err": "Conflicted booking with previous bookings.", "conflict_bookings": conflict, "status": "failed"}, 409)
+            else:
+                insertData = []
+                for i in range(formData["repeat"]):
+                    insertData.append(formData.copy())
+                    formData["startTime"] += 7 * 24 * 60 * 60
+                    formData["endTime"] += 7 * 24 * 60 * 60
```
Comment (cjunxiang) [inline backend/FacilityBooking/FacilitiesAPI.py:331]: try to use global vars or abstract to helper methods to make main business logic cleaner

**Comment by angnobel** [comment]
{
    "facilityID": 6,
    "eventName": "Test",
    "userID": "{{userStudentID}}",
    "startTime": 1680083200,
    "endTime": 	1680086800,
    "description": "Test",
    "ccaID": 1,
    "bookUntil": 1685083200,
    "forceBook": true
}

Causing 500 error

**Review by angnobel — APPROVED** [review]
LGTM

**Pull request merged**